packages/engine/src/simulation/engine.rs

```diff
@@ -249,10 +250,11 @@ impl Engine {
         self.handle_messages(state, &message_map)?;
         let message_pool = self.finalize_agent_messages(state, context)?;
         let agent_pool = self.finalize_agent_state(state, context)?;
-        let state_view = StatePools {
+        let mut state_view = StatePools {
             agent_pool,
             message_pool,
         };
+        state_view.write()?.maybe_reload()?;
```

Comment:
Why would we reload here?

Reply:
`finalize_agent_messages` and `finalize_agent_state` write to agent and message batches, but writing doesn't automatically mean that they are reloaded, so they should be reloaded here, before context packages read from them.

More simply, if we don't try reloading here, we'll later get an error about the loaded metaversion being too old. If the batches are already up to date, then `maybe_reload` won't do anything.

Reply:
I think my intuition isn't quite in place with how these methods work, and perhaps that's not something we can easily convey through program structure but lets still see if we can make it clearer. My thoughts kind of go like this:

`finalize_agent_messages` and `finalize_agent_state` return new objects, so it doesn't really imply that the data those objects represent lives somewhere else. We then very simply wrap those objects in `StatePools`, so it's not doing any other functionality. And then for some reason we have to call `reload` on a brand new plain struct.. It's quite counter-intuitive.

The docs also suggest that this is making a new copy of the data:
>    /// 4) A static dataframe is created by copying the state (dynamic) dataframe.

So why would we need to reload it if it's something new?

Reply:
I see how it can be confusing. Perhaps `maybe_reload` should be called `maybe_reload_record_batch`. In itself, a batch struct is just a reference to the actual data in shared memory. A batch can be created by creating a memory segment and writing Arrow bytes into it, without actually loading the bytes. But to access individual agents/columns in context packages, we need to load metadata about the columns, i.e. a Rust `RecordBatch`.

Reply:
Yeah it seems very strange that we'll manually construct a RecordBatch in memory. (As in creating a new one in parts, e.g. column by column) I get modifying one in place, but it certainly seems strange that we construct one rather than making a complete `RecordBatch` and putting that into memory.

Reply:
I don't see a way we can easily clarify these things at the moment though so I'm going to resolve this set of comments